### ISSUE TITLE
Train a baseline classifier

### ISSUE THREAD

**Issue by marco-c** [marco-c/autowebcompat#279]
We need to find some good options for the classifier to reach a baseline acceptable accuracy:
- [ ] Try vgg16, pretrained on imagenet
- [ ] Try vgg16, training from scratch
- [ ] Try vgg16, pretrained with pretrain.py

We can start with the RMSProp optimizer.

**Comment by sdv4** [comment]
Network - vgg16
Pretrained - none
Optimiser - sgd
Epochs - 50
Test Set Accuracy - 77.14%

Class balance: 
Y: 53%
D+N : 47%

Confusion matrix:
[[201 101]
[42 264]]

[0af4373f3246_18_59_2018_12_10.txt](https://github.com/marco-c/autowebcompat/files/2665313/0af4373f3246_18_59_2018_12_10.txt)

**Comment by sdv4** [comment]
Network - vgg16
Pretrained - pretrain.py
Optimiser - sgd
Epochs - 50
Test Set Accuracy - 45.98%

**Comment by marco-c** [comment]
What exactly did you do in the pretrain.py case?

**Comment by marco-c** [comment]
Also, could you try using rmsprop instead of sgd?

**Comment by marco-c** [comment]
> Also, could you try using rmsprop instead of sgd?

Actually, both, to compare exactly their results.

**Comment by sdv4** [comment]
Ahh, I was thinking that this might be the case. Now it makes sense why there was no .txt file, etc. 
Do you have info on what is going on in pretrain.py; in particular the "slightly different problem (for which we know the solution)"?

**Comment by marco-c** [comment]
This explains it all:
https://github.com/marco-c/autowebcompat/blob/master/pretrain.py#L58

The goal of the classifier in the pretrain case is to detect when two screenshots belong to the same website.

**Comment by sdv4** [comment]
> This explains it all:
> https://github.com/marco-c/autowebcompat/blob/master/pretrain.py#L58

Indeed it does, thanks!

**Comment by sdv4** [comment]
Class balance:
Y: 53%
D+N : 47%

Network - vgg16
Pretrained - imagenet 
image size: (224,224)
Optimiser - sgd
Epochs - 50
Test Set Accuracy - 84.9%
N+D Prediction Precision: 85.5%
Confusion matrix:
[[254 47]
[43 264]]

[tensorflow-1-vm_21_49_2018_12_20.txt](https://github.com/marco-c/autowebcompat/files/2700848/tensorflow-1-vm_21_49_2018_12_20.txt)

note: I was able to train on the full size images by using GCP instead of Colab, but I am working on an idea for fixing the memory issue in issue #280

**Comment by sdv4** [comment]
@marco-c I also performed hyper-parameter optimization with random search, and [Hyperband](https://arxiv.org/abs/1603.06560). The best configuration found was via random search:

Network - vgg16
Pretrained - no
Test Set Accuracy - 80.1%
**N+D Prediction Precision: 88.1%**
Image size: (48,48)
Optimizer: Adam
Learning rate: 1.7e-5  
Decay: 1e-6
Momentum: 0.74  
Epsilon: 8.4e-8  
fc1 L2 regularization strength: 2.32e-2  
fc2 L2 regularization strength: 5.19e-3  
fc1 dropout: 3.37e-7  
fc2 dropout: 1.71e-7

**Comment by marco-c** [comment]
Not bad! We should improve the labeling to have better/more precise results
